File: ll/array2ll.cpp
#include<iostream>
using namespace std;
class Node{
    public:
        int data;
        Node * next;
    public:
        Node(int data1,Node* next1)
        {
            data=data1;
            next=next1;
        }
        Node(int data1)
        {
            data= data1;
            next = nullptr;
        }
};
int main()
{
    //converting array to linked list
    
    int arr[] = {1,2,4,5,6,7,6,5};
    int length = sizeof(arr) / sizeof(arr[0]);
    Node *temp=nullptr;
    Node *head;
    for(int i =0;i<length;i++)
    {
        if(i==0)
        {
            head = new Node(arr[i]);
            temp=head;
        }
        else
        {
            Node *node = new Node(arr[i],temp);
            temp=node;
        }
    }
    while (temp->next!=nullptr)
    {
        cout<<temp->data;
        temp=temp->next;
    }
    return 0;
}

File: single ll/array2ll.java
class Node {
    int data;
    Node next;

    Node(int data, Node next) {
        this.data = data;
        this.next = next;
    }

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

public class array2ll {
    public static void main(String args[]) {
        int arr[] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };

        Node head = new Node(arr[0]);
        Node temp=head;
        for (int i = 1; i < arr.length; i++) {
            Node node = new Node(arr[i]);
            temp.next=node;
            temp=node;
        }
        temp=head;
        while(temp!=null)
        {
            System.out.println(temp.data);
            temp=temp.next;
        }
    }
}
